libs/mrwebrtc/src/data_channel.cpp

```diff
@@ -96,11 +112,16 @@ void DataChannel::OnStateChange() noexcept {
 
 void DataChannel::OnMessage(const webrtc::DataBuffer& buffer) noexcept {
   std::lock_guard<std::mutex> lock(mutex_);
-  if (message_callback_) {
-    message_callback_(buffer.data.data(), buffer.data.size());
+  if (buffer.binary) {
+    if (message_callback_) {
+      message_callback_(buffer.data.data(), buffer.data.size());
+    }
+  }else if (text_message_callback_) {
+    text_message_callback_(buffer.data.data(), buffer.data.size());
   }
```

Comment:
This is where we change how the library works.
Before we would always fire the message_callback_, but now it is only done when we have binary data.

I am thinking, if we want this to work just as it was before, we should maybe add the following logic,
if text_message_callback_ is not registered, always fire the message_callback. This would ensure that old implementations are still working.

if text_message_callback_ is registered, then we can split on binary data and fire the different events.

I feel it is some strange logic, but it would keep the library working just as it had before my change. What do you think @djee-ms ?